gateway/plugin/external.go

```diff
@@ -111,12 +113,28 @@ func (s ExternalServiceQuery) GetReplicas(serviceName string) (handlers.ServiceQ
 				maxReplicas = uint64(labelValue)
 			}
 		}
+
+		if len(scaleFactor) > 0 {
```

Comment:
Let's extract this to a function of its own that can take a default value as an override.

Reply:
It has more "logic" attached to it than the other labels, so this definitely makes sense.  I will go ahead and move it into a function.

Reply:
I will go ahead and do that change. It makes sense.